testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/OidcBaseTest.java

```diff
@@ -0,0 +1,295 @@
+    }
+
+    @Test
+    @OperateOnDeployment(WRONG_SECRET_APP)
+    public void testWrongClientSecret() throws Exception {
+        loginToApp(WRONG_SECRET_APP, org.wildfly.test.integration.elytron.oidc.client.KeycloakConfiguration.ALICE, org.wildfly.test.integration.elytron.oidc.client.KeycloakConfiguration.ALICE_PASSWORD, HttpURLConnection.HTTP_FORBIDDEN, null);
+    }
+
+    public static void loginToApp(String appName, String username, String password, int expectedStatusCode, String expectedText) throws Exception {
+        loginToApp(appName, username, password, expectedStatusCode, expectedText, true);
+    }
+
+    public static void loginToApp(String appName, String username, String password, int expectedStatusCode, String expectedText, boolean loginToKeycloak) throws Exception {
+        final URI requestUri = new URL("http", TestSuiteEnvironment.getHttpAddress(), TestSuiteEnvironment.getHttpPort(),
+                "/" + appName + SimpleSecuredServlet.SERVLET_PATH).toURI();
+        CookieStore store = new BasicCookieStore();
+        HttpClient httpClient = TestHttpClientUtils.promiscuousCookieHttpClientBuilder()
+                .setDefaultCookieStore(store)
+                .setRedirectStrategy(new LaxRedirectStrategy())
+                .build();
+        HttpGet getMethod = new HttpGet(requestUri);
+        HttpContext context = new BasicHttpContext();
+        HttpResponse response = httpClient.execute(getMethod, context);
+        try {
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (loginToKeycloak) {
+                assertTrue("Expected code == OK but got " + statusCode + " for request=" + requestUri, statusCode == HttpURLConnection.HTTP_OK);
+                Form keycloakLoginForm = new Form(response);
+                HttpResponse afterLoginClickResponse = simulateClickingOnButton(httpClient, keycloakLoginForm, username, password, "Sign In");
+                afterLoginClickResponse.getEntity().getContent();
+                assertEquals(expectedStatusCode, afterLoginClickResponse.getStatusLine().getStatusCode());
+                if (expectedText != null) {
+                    String responseString = new BasicResponseHandler().handleResponse(afterLoginClickResponse);
+                    assertTrue(responseString.contains(expectedText));
+                }
+            } else {
+                assertTrue("Expected code == FORBIDDEN but got " + statusCode + " for request=" + requestUri, statusCode == HttpURLConnection.HTTP_FORBIDDEN);
+            }
+        } finally {
+            HttpClientUtils.closeQuietly(response);
+        }
+    }
+
+    public static class KeycloakSetup implements ServerSetupTask {
+
+        @Override
+        public void setup(ManagementClient managementClient, String containerId) throws Exception {
+            assumeTrue("Docker isn't available, OIDC tests will be skipped", isDockerAvailable());
```

Comment:
@darranl I think @fjuma  wanted to ignore the tests on Windows. She mentioned privately that the testcontainers instance she is using to obtain the Keycloak server isn't supported on Windows. I thinks this assumeTrue is here for that reason. But the exception thrown by the assumeTrue is caught in the [ServerSetupObserver.java](https://github.com/wildfly/wildfly-arquillian/blob/main/common/src/main/java/org/jboss/as/arquillian/container/ServerSetupObserver.java#L187) , logged and ignored. I will create a commit that runs this assumeTrue in BeforeClass method or at some other suitable location.

Reply:
+1